docs/adr/adr-013-blocksync-overhaul-part-2.md

```diff
@@ -0,0 +1,535 @@
+# ADR #013: Block Data Sync Overhaul: Part II - Network
+
+## Authors
+
+@Wondertan @renaynay
+
+> This ADR took a while to get through all the necessary details. Special thanks to awesome @renaynay for dozens of
+> deep dive calls, note-taking, thoughts, and ideas.
+> Similarly, the time spent on the experiments with the format of the documents was refactored three times.
+
+## Glossary
+
+- LN - Light Node
+- FN - Full Node
+- BN - Bridge Node
+- Core Network - Consensus Network of Validators/Proposers producing new Celestia Blocks
+- [EDS(Extended Data Square)](https://github.com/celestiaorg/rsmt2d/blob/80d231f733e9dd8ca166c3d670470ed9a1c165d9/extendeddatasquare.go#L13) - plain Block
+  data omitting headers and other block metadata
+- ODS - Original Data Square or the first quadrant of the EDS. It contains real user data and padding
+- Recent EDS - Newly produced EDS coming from the Core consensus network
+- ND - Namespaced data/shares. Constitutes EDS
+
+## Overview
+
+This document is a continuation of the BlockSync Overhaul design. The previous document(TODO: Link) described the design
+for the storage subsystem of the new block synchronization protocol of Celestia's Data Availability network and the Part
+II focuses on networking.
+
+The new design drastically speeds up EDS retrieval in __happy path__ by streaming ODS as a single file and recomputing
+parity data forming EDS. A similar approach is taken for ND retrieval. The main goal is to make EDS and ND retrieval
+consistently outpace EDS production for the max EDS size for historical and recent EDSes.
+
+The existing EDS retrieval logic(TODO: Link something) remains a __fallback path__ if the __happy path__ fails due to a
+lack of connected peers or the coordinated data withholding attack. With the __fallback path__, retrieval leverages all 
+connected peers, including LNs, and granularly requests EDS pieces - __shares__, increasing the likelihood of EDS 
+reconstruction while paying a slower retrieval time price.
```

Comment:
```suggestion
The [existing EDS retrieval
logic](https://github.com/celestiaorg/celestia-node/blob/7cc511d74bead176e39312798e61a8c83ac2fb7f/share/eds/retriever.go)
remains as a __fallback path__ if the __happy path__ fails. This could occur due to a lack of
connected peers or a coordinated data withholding attack. With the __fallback path__, retrieval
leverages all connected peers, including LNs, and granularly requests EDS pieces - __shares__,
increasing the likelihood of EDS reconstruction while paying a slower retrieval time price.
```